### ISSUE TITLE
Robustness of certain integration tests.

### ISSUE THREAD

**Issue by AaronSpieler** [awslabs/gluon-ts#764]
## Description
Some tests are not robust and often fail:
* test/model/deepstate/test_model.py
* test/model/wavenet/test_model.py

They should be made more robust. 
Currently they fail in unrelated PRs which creates unnecessary noise.

**Comment by jaheba** [comment]
I think we had a retry-logic for them before. Which of course makes them pass more often, but is nothing I would deem as "robust".

We could relax the requirements or train longer (with negative effects on training time, and thus test-time).

Alternatively, we could create regression tests, which run independently.

**Comment by lostella** [comment]
> Alternatively, we could create regression tests, which run independently.

Yes to that, of course, and then turning the current tests into very quick trainings and very quick prediction tasks with no accuracy assertion (essentially smoke tests).

**Comment by arangatang** [comment]
So my plan to solve this issue is to calculate the standard deviation and mean of the accuracy of the two models which are causing the tests to fail. 
Thereafter i'll change the current accuracy test for each model to instead check that the models accuracy (when testing) is between a certain confidence interval of the mean. 

Furthermore I'll edit the `.github/workflows/tests.yml` [link](https://github.com/awslabs/gluon-ts/blob/master/.github/workflows/tests.yml) to run these tests in a separate environment/container when running the tests on GitHub.

Ill also make smoke tests similar to the one available [here](https://github.com/awslabs/gluon-ts/blob/master/test/model/deepstate/test_deepstate_smoke.py)

I do have one question:

1. what confidence interval should we use/p value, is p=0.05 good?

Are there any concerns with this proposed solution or something you want to change?

**Comment by lostella** [comment]
So I think we should have two sets of tests, right?
1. Smoke tests: run very short training on whatever data, testing that the resulting predictor behaves as expected. This can be similar to [this one for deepstate](https://github.com/awslabs/gluon-ts/blob/master/test/model/deepstate/test_deepstate_smoke.py#L121-L127), but could include other things: for example, one could assert that the forecast start date and length is as expected. No accuracy should be tested here.
2. Regression tests: train a model on a real-world dataset, get accuracy metrics and verify that they are below some expected value. These tests will run longer, so ideally they are run separately from whatever `pytest` executes by default.

I think these are two separate stories?

**Comment by arangatang** [comment]
> So I think we should have two sets of tests, right?

Yeah that's my understanding to

> 2. Regression tests: train a model on a real-world dataset, get accuracy metrics and verify that they are below some expected value. These tests will run longer, so ideally they are run separately from whatever `pytest` executes by default.

Running them separately is easily done with minor modifications of [this](https://github.com/awslabs/gluon-ts/pull/782) pull request. Thus I will start with this task then.

> I think these are two separate stories? @jaheba

I think @jaheba is out of office today

**Comment by lostella** [comment]
As per offline discussion: this story is about making the tests mention in the original post less noisy, i.e. fail less often, but keeping them fast enough that they can run with every build or local `pytest` run.

Proper model accuracy regression testing is tracked by #379

**Comment by arangatang** [comment]
I believe the flaky behaviour of 
* test/model/wavenet/test_model.py
was resolved in #793 as a sideffect of increasing the accuracy value a bit + fixing the timeout issue itself. 

Until the large regression test is in place from #379 should we increase the accuracy limit also for the test below as a temporary work around?

* test/model/deepstate/test_model.py
